lib/metadata/bucketfile/backend.js

```diff
@@ -29,152 +21,41 @@ class BucketFileInterface {
     constructor() {
         this.logger = logger;
         if (cluster.isMaster) {
-            this.startServer();
-        } else {
-            this.refcnt = 0;
-            this.waitreco = 0;
-            this.realReConnect();
-            this.notifier = new events.EventEmitter();
+            this.mdServer = new MetadataServer(
+            { metadataPath: METADATA_PATH,
+              metadataPort: METADATA_PORT,
+              log: config.log });
+            this.mdServer.startServer();
+        }
+        this.mdClient = new MetadataClient(
+            { metadataHost: 'localhost',
+              metadataPort: METADATA_PORT,
+              log: config.log });
+        this.mdDB = this.mdClient.openDB();
+        this.metastore = this.mdDB.openSub(METASTORE);
+        if (cluster.isMaster) {
```

Comment:
Correct me if I'm wrong, the conditional is still necessary because we did not change the architecture yet, the server is still spawned locally on the same container, so it has to be unique, and one way of doing it (that was there before) is to check that we're the master process in the cluster.

The proper separation of the server in a separate daemon will be the topic of another PR :)